libtelemetry/src/main/java/com/mapbox/services/android/telemetry/TelemetryClient.java

```diff
@@ -48,6 +48,10 @@ public void sendEvent(Event event, Callback callback) throws IOException {
     sendBatch(oneEvent, callback);
   }
 
+  void updateDebugLoggingEnabled(boolean debugLoggingEnabled) {
+    setting = setting.toBuilder().debugLoggingEnabled(debugLoggingEnabled).build();
```

Comment:
`setting.toBuilder()` will enter in an infinite loop which will cause a `StackOverflowError` exception 💥 at runtime.

https://github.com/mapbox/mapbox-events-android/blob/2d9ac6b517aab5db6205d27d8765fc0c0ca65004/libtelemetry/src/main/java/com/mapbox/services/android/telemetry/TelemetryClientSettings.java#L77-L79